e2e/app/component/annotation-field-modal.ts

```diff
@@ -0,0 +1,59 @@
+import {Page} from 'puppeteer';
+import {savePageToFile, takeScreenshot} from 'utils/save-file-utils'
+import {LinkText} from 'app/text-labels';
+import Button from 'app/element/button';
+import Modal from './modal';
+import ReactSelect from 'app/element/react-select';
+import {waitWhileLoading, waitForText} from 'utils/waits-utils';
+
+const modalTitle = 'Create a Review-Wide Annotation Field';
+
+export enum AnnotationType {
+    FreeText = 'Free Text',
+    DropdownList = 'Dropdown List',
+    Date = 'Date',
+    TrueFalseCheckBox ='True/False CheckBox',
+    NumericField = 'Numeric Field'
+  }
+
+export default class AnnotationFieldModal extends Modal {
+
+  constructor(page: Page, xpath?: string) {
+    super(page, xpath);
+  }
+
+  async waitForLoad(): Promise<this> {
+    try {
+      await super.waitForLoad();
+      await waitForText(this.page, modalTitle, {xpath: this.getXpath()});
+    } catch (e) {
+      await savePageToFile(this.page);
+      await takeScreenshot(this.page);
+      throw new Error(`AnnotationFieldModal waitForLoad() encountered ${e}`);
+    }
+    return this;
```

Comment:
put `await super.waitForLoad` outside of try block because it already takes screenshot when fails.

Reply:
should I put it before  `return this;`?

Reply:
before `try`

Reply:
Ok!

Reply:
done!